### ISSUE TITLE
Doctest directives and comments missing from code samples

### ISSUE THREAD

**Issue by stevendaprano** [python/cpython#80856]
BPO | [36675](https://bugs.python.org/issue36675)
--- | :---
Nosy | @terryjreedy, @gpshead, @ezio-melotti, @merwok, @stevendaprano, @jdemeyer, @willingc, @asottile, @JulienPalard, @jdlh, @csabella, @miss-islington
PRs | <li>python/cpython#13125</li><li>python/cpython#16024</li><li>python/cpython#16030</li><li>python/cpython#23620</li>
Files | <li>[missing_directives.png](https://bugs.python.org/file48277/missing_directives.png "Uploaded as image/png at 2019-04-20.10:43:50 by @stevendaprano")</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2019-04-20.10:43:50.531>
labels = ['3.8', 'type-bug', '3.7', 'docs']
title = 'Doctest directives and comments missing from code samples'
updated_at = <Date 2021-10-26.04:00:41.470>
user = 'https://github.com/stevendaprano'
```

bugs.python.org fields:
```python
activity = <Date 2021-10-26.04:00:41.470>
actor = 'minghua'
assignee = 'docs@python'
closed = False
closed_date = None
closer = None
components = ['Documentation']
creation = <Date 2019-04-20.10:43:50.531>
creator = 'steven.daprano'
dependencies = []
files = ['48277']
hgrepos = []
issue_num = 36675
keywords = ['patch']
message_count = 17.0
messages = ['340570', '340959', '342648', '342651', '342652', '342655', '342664', '344714', '352093', '352120', '352128', '367173', '383075', '383732', '384361', '385632', '385661']
nosy_count = 14.0
nosy_names = ['terry.reedy', 'gregory.p.smith', 'ezio.melotti', 'eric.araujo', 'steven.daprano', 'docs@python', 'jdemeyer', 'willingc', 'Anthony Sottile', 'mdk', 'JDLH', 'cheryl.sabella', 'miss-islington', 'minghua']
pr_nums = ['13125', '16024', '16030', '23620']
priority = 'normal'
resolution = 'later'
stage = None
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue36675'
versions = ['Python 2.7', 'Python 3.7', 'Python 3.8']
```

</p></details>

**Comment by sorcio** [comment]
> Due to #24282 this is sadly un-fixed.
> 
> Either we find another way to fix this, either we wait 3 releases and we re-commit #23620.

Since docs are now built on Sphinx 4.5.0, it might be a good time to re-fix this.

I applied a cherry-picked https://github.com/python/cpython/pull/23620/commits/eb7fcb8dbcaf83794b10e7e9802da6ec68e0cedd locally and it seems to work fine.

Made this into a new PR https://github.com/python/cpython/pull/92318

**Comment by ezio-melotti** [comment]
This is now fixed on 3.10/3.11/main.
Thanks for the PR!
